modules/popup/react/README.md

```diff
@@ -124,3 +257,90 @@ Default:
 
 > A ref to the underlying popup container element. Use this to check click targets against when
 > closing a popup.
+
+## Hooks
+
+### usePopupStack
+
+```ts
+usePopupStack(ref: React.RefObject<HTMLElement>): void
+```
+
+This hook will add a `ref` element to the Popup stack on mount and remove on unmount. If you use
+`Popper`, the popper `ref` is automatically added/removed from the Popup stack. The Popup stack is
+required for proper z-index values to ensure Popups are rendered correct. It is also required for
+global listeners like click outside or escape key closing a popup. Without the Popup stack, all
+popups will close rather than only the topmost one.
+
+This should be used by all stacked UIs unless using the `Popper` component.
+
+## useAssistiveHideSiblings
+
+```ts
+useAssistiveHideSiblings(ref: React.RefObject<HTMLElement>): void
+```
+
+This hook will hide all sibling elements from assistive technology. Very useful for modal dialogs.
+This will set `aria-hidden` for sibling elements of the provided `ref` element and restore the
+previous `aria-hidden` to each component when the component is unmounted. For example, if added to a
+Modal component, all children of `document.body` will have an `aria-hidden=true` applied _except_
+for the provided `ref` element (the Modal). This will effectively hide all content outside the Modal
+from assistive technology including Web Rotor for VoiceOver for example.
+
+**Note**: The provided `ref` element should be root element of your component so that other elements
+_outside_ your component will be hidden rather than elements _inside_ your component.
+
+This should be used on stacked UI elements that need to hide content. Like Modals.
+
+## useBringToTopOnClick
+
+```ts
+useBringToTopOnClick(ref: React.RefObject<HTMLElement>): void
+```
+
+This hook will bring an element to the top of the stack when any element inside the provided `ref`
+element is clicked. If `Popper` was used or `PopupStack.add` provided an `owner`, all "child" popups
+will also be brought to the top. A "child" popup is a Popup is a Popup that was opened from another
+Popup. Usually this is a Tooltip or Select component inside something like a Modal.
+
+This should be used on stacked UI elements that are meant to persist, like Windows.
+
+## useCloseOnEscape
+
+```ts
+useCloseOnEscape(ref: React.RefObject<HTMLElement>, onClose: () => void): void
```

Comment:
Many of these don't return `void` - they return a function to remove the listeners

Reply:
I think you're talking about callback functions given to `use[Layout]Effect`. The callback here is usually an event listener or a callback function that takes no arguments. For example, `handleClose` in `Popup.tsx` is `() => void`.

This is the definition of a `MouseEvent` in React:
```ts
type React.MouseEventHandler<T = Element> = (event: MouseEvent<T, globalThis.MouseEvent>) => void
```

There is nothing to capture the return value of the callback.

Reply:
Whoops - you're right. For some reason I missed the `use[Layout]Effect` when I saw the return statement and commented here without really thinking things through. You're good 👍